src/vm/jithelpers.cpp

```diff
@@ -2313,6 +2317,9 @@ BOOL ObjIsInstanceOf(Object *pObject, TypeHandle toTypeHnd, BOOL throwCastExcept
 
     BOOL fCast = FALSE;
 
+    // when castability is decided by calling to actual objects, do not cache
+    BOOL canCache = TRUE;
```

Comment:
+X Bytes per BOOL per instance just for the canCache seems like overkill here... especially considering we have fCast.

Nit: size_t or uint32 would be better here and allow more flags to be used rather than just on or off.